core/metamodel/src/main/java/org/apache/isis/core/metamodel/services/appfeat/ApplicationFeatureRepositoryDefault.java

```diff
@@ -64,19 +68,23 @@
 import lombok.extern.log4j.Log4j2;
 
 @Service
-@Named("isis.metamodel.ApplicationFeatureRepositoryDefault")
+@Named(ApplicationFeatureRepositoryDefault.LOGICAL_TYPE_NAME)
 @Log4j2
 public class ApplicationFeatureRepositoryDefault
 implements ApplicationFeatureRepository, MetamodelListener {
 
+    static final String LOGICAL_TYPE_NAME = IsisModuleCoreMetamodel.NAMESPACE + ".ApplicationFeatureRepositoryDefault";
+
     // -- caches
     private Map<String, ApplicationFeatureId> featureIdentifiersByName;
     final SortedMap<ApplicationFeatureId, ApplicationFeature> namespaceFeatures = _Maps.newTreeMap();
     private final SortedMap<ApplicationFeatureId, ApplicationFeature> typeFeatures = _Maps.newTreeMap();
+    private final SortedMap<LogicalType, ApplicationFeatureId> typeFeatureIdByLogicalType = _Maps.newTreeMap();
     private final SortedMap<ApplicationFeatureId, ApplicationFeature> memberFeatures = _Maps.newTreeMap();
     private final SortedMap<ApplicationFeatureId, ApplicationFeature> propertyFeatures = _Maps.newTreeMap();
     private final SortedMap<ApplicationFeatureId, ApplicationFeature> collectionFeatures = _Maps.newTreeMap();
     private final SortedMap<ApplicationFeatureId, ApplicationFeature> actionFeatures = _Maps.newTreeMap();
+    private final SortedMap<String, List<ApplicationFeature>> memberFeaturesByLogicalTypeName = _Maps.newTreeMap();
```

Comment:
this is a multi-map ... I think I should add a **TreeMap of ArrayLists** variant to `_MultiMaps`, which we currently don't have it seems